# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/pratham6306/ml-internship-flyrank/blob/main/work/notebooks/w01_research_question.ipynb)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

# My Lane

I chose **Lane 2: Refresh / Content Opportunity Scoring**.

This lane focuses on helping SEO specialists and content editors decide which webpages should be reviewed first for refresh or optimization. The starter dataset contains observable search and engagement signals such as impressions, clicks, CTR, sessions, and content age, making it suitable for prioritizing content review. The goal is to support human decision-making rather than automatically deciding which pages must be refreshed.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

# Research Question

**Decision**

Which webpages should be reviewed first for content refresh?

**Who acts on the output?**

SEO specialists or content editors.

**Action**

Review the highest-ranked pages and decide whether to refresh content, improve metadata, expand content, or monitor performance.

**Unit of Analysis**

One webpage.

**Output**

A ranked list of webpages with a refresh opportunity score and supporting reason codes.

**Cost of a Wrong Recommendation**

Incorrect recommendations may waste editorial effort or overlook high-value pages that deserve attention.

**Why Data or ML Helps**

Large websites contain many webpages, making manual prioritization difficult. Observable search and engagement signals can be combined to produce a more useful review queue while still requiring human judgment.


## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [10]:
!git clone https://github.com/pratham6306/ml-internship-flyrank.git

fatal: destination path 'ml-internship-flyrank' already exists and is not an empty directory.


In [11]:
!ls ml-internship-flyrank/data/raw
import pandas as pd

df = pd.read_csv("/content/ml-internship-flyrank/data/raw/content_refresh_anonymized.csv")

print(f"Rows: {len(df)}")
print(f"Columns: {len(df.columns)}")

corr = df["search_volume"].corr(df["impressions_90d"])
print(f"Correlation between search_volume and impressions_90d: {corr:.3f}")

print("\nMedian Impressions:", df["impressions_90d"].median())
print("Median Sessions:", df["sessions_90d"].median())

content_refresh_anonymized.csv
Rows: 30000
Columns: 44
Correlation between search_volume and impressions_90d: 0.001

Median Impressions: 731.0
Median Sessions: 7.0


### Quick Look at the Data

The starter dataset contains **30,000 webpages** described by **44 features**, providing enough observations to explore patterns related to content performance.

The correlation between **search_volume** and **impressions_90d** is **0.001**, indicating that keyword search volume alone does not explain how much traffic a page receives. This suggests that prioritizing pages for review requires combining multiple observable signals rather than relying on a single metric.

The median webpage received **731 impressions** but only **7 sessions** over the previous 90 days. This highlights that visibility does not always translate into user engagement, making content prioritization a meaningful decision-support problem.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

# Careful Claims

This work identifies patterns in historical search and engagement data to support content review decisions.

The analysis does not prove that refreshing a webpage will improve its performance, nor does it claim to predict Google's ranking algorithm.

The recommendations are intended as decision-support for human reviewers based on observed historical data.

## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.